langchain/src/vectorstores/weaviate.ts

```diff
@@ -164,12 +164,22 @@ export class WeaviateStore extends VectorStore {
     });
 
     try {
-      await this.client.batch
+      const responses = await this.client.batch
         .objectsBatcher()
         .withObjects(...batch)
         .do();
+      // if storing vectors fails, we need to know why
+      const errorMessages: string[] = [];
+      responses.forEach((response) => {
+        if (response?.result?.errors?.error) {
+          errorMessages.push(...response.result.errors.error.map((err) => err.message ?? ""));
+        }
+      });
+      if (errorMessages.length > 0) {
+        throw new Error(errorMessages.join("\n"));
```

Comment:
Might be good to have some base case here if no messages can be parsed - e.g. `Unspecified error`

Reply:
Yep, I didn't really think about a sensible fallback in line 175 and just took the empty string to fulfil typing restrictions.
Anyways, just preparing for this possibility I'd change the fallback in line 175 from empty string to `"!! Unfortunately no error message was presented in the API response !!"`. 
Commit follows.